genesyscloud/architect_grammar/resource_genesyscloud_architect_grammar.go

```diff
@@ -0,0 +1,189 @@
+package architect_grammar
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v112/platformclientv2"
+	"log"
+	gcloud "terraform-provider-genesyscloud/genesyscloud"
+	resourceExporter "terraform-provider-genesyscloud/genesyscloud/resource_exporter"
+	"terraform-provider-genesyscloud/genesyscloud/util/resourcedata"
+	"time"
+)
+
+/*
+The resource_genesyscloud_architect_grammar.go contains all the methods that perform the core logic for a resource.
+*/
+
+// getAllAuthArchitectGrammar retrieves all of the architect grammars via Terraform in the Genesys Cloud and is used for the exporter
+func getAllAuthArchitectGrammar(ctx context.Context, clientConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	proxy := getArchitectGrammarProxy(clientConfig)
+	resources := make(resourceExporter.ResourceIDMetaMap)
+
+	grammars, err := proxy.getAllArchitectGrammar(ctx)
+	if err != nil {
+		return nil, diag.Errorf("Failed to get grammars: %v", err)
+	}
+
+	for _, grammar := range *grammars {
+		log.Printf("Dealing with grammar id : %s", *grammar.Id)
```

Comment:
I wouldn't bother with this log statement. It can complicate things a bit when you're exporting a lot of objects